calicoctl/commands/node/run.go

```diff
@@ -85,6 +92,29 @@ Options:
                            node resource.  If there is no configured value
                            and the --ip6 option is omitted, the node will not
                            route IPv6.
+     --ip-autodetection-method=<IP_AUTODETECTION_METHOD>
+                           Specify the autodetection method for detecting the
+                           local IPv6 routing address for this node.  The valid
+                           options are:
+                           > first-found
+                             Use the first valid IP address on the first
+                             enumerated interface (common known exceptions are
+                             filtered out, e.g. the docker bridge).  It is not
+                             receommended to use this if you have multiple
+                             external interfaces on your host.
+                           > can-reach=<IP OR DNS>
```

Comment:
It's more bash like if you don't use a "=" after the parameter, e.g.:

`--ip-autodetection-method can-reach=10.0.0.1`

It's similar to the --output method where you can specify parameters on some options.   I believe k8s follows the same approach.

Personally, I think this is OK.